module/executiondatasync/pruner/pruner.go

```diff
@@ -0,0 +1,206 @@
+package pruner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/executiondatasync/tracker"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/util"
+)
+
+const (
+	defaultHeightRangeTarget = uint64(400_000)
+	defaultThreshold         = uint64(100_000)
+)
+
+// Pruner is a component responsible for pruning data from
+// execution data storage. It is configured with the following
+// parameters:
+// * Height range target: The target number of most recent blocks
+//   to store data for. This controls the total amount of data
+//   stored on disk.
+// * Threshold: The number of block heights that we can exceed
+//   the height range target by before pruning is triggered. This
+//   controls the frequency of pruning.
+// The Pruner consumes a stream of fulfilled height notifications,
+// and triggers pruning once the difference between the fulfilled
+// height and the last pruned height reaches the height range
+// target + threshold.
```

Comment:
```suggestion
// The Pruner tries to preserve at minimum `heightRangeTarget` number of height of data in storage.
// It's notified with unpruned notifications, when there are `heightRangeTarget + threshold` number 
// of unpruned height, it prunes the storage until there are `heightRangeTarget` unpruned height left.
```

Reply:
I don't think this is clearer, it's just a different way of saying the same thing